payu/models/staged_cable.py

```diff
@@ -0,0 +1,205 @@
+"""payu.models.cable
+   ================
+
+   Driver interface to CABLE-POP_TRENDY branch
+
+   :copyright: Copyright 2021 Marshall Ward, see AUTHORS for details
+   :license: Apache License, Version 2.0, see LICENSE for details
+"""
+
+# Standard Library
+import os
+import shutil
+
+# Extensions
+import f90nml
+import yaml
+# import xarray
+
+# Local
+from payu.models.model import Model
+
+
+def deep_update(d_1, d_2):
+    """Deep update of namelists."""
+    for key, value in d_2.items():
+        if isinstance(value, dict):
+            deep_update(d_1[key], d_2[key])
+        else:
+            d_1[key] = value
+
+
+class StagedCable(Model):
+    """A driver for running staged CABLE spin-up configurations."""
+
+    def __init__(self, expt, name, config):
+        super(StagedCable, self).__init__(expt, name, config)
+
+        self.model_type = 'staged_cable'
+        self.default_exec = 'cable'
+
+        self.config_files = ['stage_config.yaml']
+        self.optional_config_files = ['cable.nml', 'cru.nml',
+                                      'luc.nml', 'met_names.nml']
+        self.configuration_log = {}
+
+        # Read the stage_config.yaml file
+        with open('stage_config.yaml', 'r') as stage_conf_f:
+            self.stage_config = yaml.safe_load(stage_conf_f)
+
+        if not os.path.isfile('configuration_log.yaml'):
+            # Build a new configuration log
+            self._build_new_configuration_log()
+        else:
+            # Read the current configuration log
+            self._read_configuration_log()
+
+        # Add the restart directories to inputs
+        num_completed_stages = len(self.configuration_log['completed_stages'])
+        for restart_dir in reversed(range(num_completed_stages)):
+            config['input'].append(os.path.realpath(
+                f'archive/output{restart_dir:03d}/restart'))
```

Comment:
I couldn't work out precisely whether this function would give me the functionality I needed i.e. what is done with the list of files returned from this function. Is it simply a list of files that ended up being added to the manifest?

Reply:
The way manifests work (in general) is that they reflect the state of whatever is in the `work` directory after all the model setup steps. This was deliberate, rather than inject manifest stuff in every time a model driver made some modification to what was in the `work` directory, just wait until it all washes out and then record the result.

So I believe `get_prior_restart_files` just gets called and then loops through the files creating symlinks into `work`

https://github.com/payu-org/payu/blob/83001a2b45ef786e178894f77dd9fbc0878e73c1/payu/models/model.py#L261-L270

Reply:
The issue I see here is that (I think) `self.prior_restart_path` points strictly the most recent output directory in the `archive`. This precludes us from pulling restarts from earlier stages, which we often do.

Reply:
I don't think there was a workable solution built around extending the ```self.get_prior_restart_files()```, because of the way it handles the files returned by this method. To have the correct file added to the manifest, files from restarts that are not the most recent would need to be prepended by ```"../restart{x}"```, which would have unintended consequences for the link location.

The solution I went with runs through the previous restart directories, from newest to oldest, and adds the files in the restart directories to the manifest if said file doesn't already exist in the manifest. This way we don't write over newer versions of a given restart, but still have the ability to mix and match restarts from different science modules.